python/cuml/svm/svc.pyx

```diff
@@ -0,0 +1,282 @@
+                             int n_rows, int n_cols, math_t *labels,
+                             const svmParameter &param,
+                             KernelParams &kernel_params,
+                             svmModel[math_t] &model) except+
+
+    cdef void svcPredict[math_t](
+        const cumlHandle &handle, math_t *input, int n_rows, int n_cols,
+        KernelParams &kernel_params, const svmModel[math_t] &model,
+        math_t *preds, math_t buffer_size, bool predict_class) except +
+
+    cdef void svmFreeBuffers[math_t](const cumlHandle &handle,
+                                     svmModel[math_t] &m) except +
+
+
+class SVC(SvmBase):
+    """
+    SVC (C-Support Vector Classification)
+
+    Currently only binary classification is supported.
+
+    The solver uses the SMO method to fit the classifier. We use the Optimized
+    Hierarchical Decomposition [1] variant of the SMO algorithm, similar to [2]
+
+    References
+    ----------
+    [1] J. Vanek et al. A GPU-Architecture Optimized Hierarchical Decomposition
+         Algorithm for Support VectorMachine Training, IEEE Transactions on
+         Parallel and Distributed Systems, vol 28, no 12, 3330, (2017)
+    [2] Z. Wen et al. ThunderSVM: A Fast SVM Library on GPUs and CPUs, Journal
+    *      of Machine Learning Research, 19, 1-5 (2018)
+        https://github.com/Xtra-Computing/thundersvm
+
+    """
+    def __init__(self, handle=None, C=1, kernel='rbf', degree=3,
+                 gamma='auto', coef0=0.0, tol=1e-3, cache_size=200.0,
+                 max_iter=-1, nochange_steps=1000, verbose=False):
+        """
+        Construct an SVC classifier for training and predictions.
+
+        Parameters
+        ----------
+        handle : cuml.Handle
+            If it is None, a new one is created for this class
+        C : float (default = 1.0)
+            Penalty parameter C
+        kernel : string (default='rbf')
+            Specifies the kernel function. Possible options: 'linear', 'poly',
+            'rbf', 'sigmoid'. Currently precomputed kernels are not supported.
+        degree : int (default=3)
+            Degree of polynomial kernel function.
+        gamma : float or string (default = 'auto')
+            Coefficient for rbf, poly, and sigmoid kernels. You can specify the
+            numeric value, or use one of the following options:
+            - 'auto': gamma will be set to 1 / n_features
+            - 'scale': gamma will be se to 1 / (n_features * X.var())
+        coef0 : float (default = 0.0)
+            Independent term in kernel function, only signifficant for poly and
+            sigmoid
+        tol : float (default = 1e-3)
+            Tolerance for stopping criterion.
+        cache_size : float (default = 200 MiB)
+            Size of the kernel cache during training in MiB. The default is a
+            conservative value, increase it to improve the training time, at
```

Comment:
I don’t think we have an exact equivalent, but it might be a good follow up to monitor usage and do some benchmarking to see if we can get an optimal (or closer to) size for the default